### ISSUE TITLE
Feature request: Add LikerID of Matters Article Author in html or GraphQL API

### ISSUE THREAD

**Issue by askender** [thematters/developer-resource#1]
This is important, because:
- People which organize an activity can easily transfer LikeCoin to the Author.
- Dapps have more chance to make innovation about economic incentives.

Besides, LikeButton iframe contain LikerID, so I think hiding LikerID is not about privacy.

**Comment by guoliu** [comment]
Thanks for submitting the issue! LikerID is included in the API, you can check it out in the [playground](https://server-test.matters.news/playground),`User.liker.likerId`. You can click on the `DOCS` tab on the right side to inspect the whole schema ;-)

**Comment by askender** [comment]
Thanks a lot! I'm a newbie of GraphQL.   And the `DOCS` in playground is useful !

(I found the article list api by `chrome inspect` before)  :(

Still have two questions:



**1**
```
query {
  user(input: {userName:"guo"}) {
    userName
    id
    liker {
      likerId
    }
  }
}
```
return   `unauthorized user for field likerId`
- how can I be authorized, will you have an example?
- why only the `liker` info need authorized?

**2**
the default endpoint is `https://server-test.matters.news/graphql` which the data is not newest.
when I  change it to `https://server.matters.news/graphql`
it returns:
```
{
  "error": "Failed to fetch. Please check your connection"
}
```
By the way, I can visit `https://server.matters.news/graphql` via chrome.

I know I can test `https://server.matters.news/graphql` using python. But if everyone can test `https://server.matters.news/graphql` on playground, it will be much better.

**Comment by guoliu** [comment]
> why only the liker info need authorized?

Ah my bad, our original security model assumes likerId to be part of user's privacy. This could be fixed soon, just need to double confirm if there is any security issues.

> the default endpoint is https://server-test.matters.news/graphql which the data is not newest

There are two end points: `xxx.matters.news/playground` for the playground UI, and `xxx.matters.news/graphql` for the API. So you can test with the former and use the queries and mutations in the latter.

There are two environments: `server.matters.news` for production, and `server-test.matters.news` for testing. It is recommended for developers to use the test environment first, then switch to production environment when ready.

Hope it helps!

**Comment by guoliu** [comment]
Liker Id is public available now, closing this issue.

**Comment by askender** [comment]
Liker Id is public available only on playground.

If I use the real endpoint: <https://server.matters.news/graphql>, still got **unauthorized user for field likerId**

query:
```
query {
  user(input: {userName:"guo"}) {
    id
    liker {
      likerId
    }
  }
}
```

It return:
```
{
    "errors": [
        {
            "message": "unauthorized user for field likerId",
            "locations": [
                {
                    "line": 5,
                    "column": 7
                }
            ],
            "path": [
                "user",
                "liker",
                "likerId"
            ],
            "extensions": {
                "code": "FORBIDDEN",
                "exception": {
                    "level": "error"
                }
            }
        }
    ],
    "data": {
        "user": {
            "id": "VXNlcjo1NTk",
            "liker": {
                "likerId": null
            }
        }
    }
}
```
@guoliu

**Comment by askender** [comment]
> /
Thanks.  Your reply is helpful.
Sorry for the late reply.
I know it is safer to using playground only for developing.

What I wish is a place which I can test the real endpoint: <https://server.matters.news/graphql>

**And I found that** the previous error(I pasted it here) when I test the real endpoint on playground is because the *block* in China
```
{
  "error": "Failed to fetch. Please check your connection"
}
```
My proxy is not useful on the playground which is very strange. Other sites such as Google and Matters is fine.
If I turn off the proxy, the test-endpoint on playgroud also return this error.


I can test the real API using python on my vps or using Postman mac app(with proxy).

**Comment by guoliu** [comment]
Liker ID is public accessible, but you still need to login and authenticate as usual. I assume you authenticate in playground but not when calling the API, make sure to include your token in header `x-access-token` :-)

Sorry I'm not familiar with how is GFW blocking playground, but I can verify that it is working outside GFW. If it's a general question about using graphql API, you can open another issue with example code. You might want to use tools such as [Apollo](https://www.apollographql.com/docs/react/) to simplify query building.

**Comment by askender** [comment]
forgot GFW, we can across it.
Thanks for your tips about `x-access-token`

I test it again and read your answer again. There is still a question

**1**
> There are two end points: xxx.matters.news/playground for the playground UI, and xxx.matters.news/graphql for the API. 
I read this sentence again. I found that I missed the information in it. There are **two playground UI** and two API.   And I cannot test `https://server.matters.news/graphql` on `https://server-test.matters.news/playground` or test `https://server-test.matters.news/graphql` on `https://server.matters.news/playground`.   Because the API **can be changed** on playground UI, it make me confused before.
This problem is finally solved.  I found the cause in Chrome console.  (sorry for my simple mistake)
```
Access to fetch at 'https://server-test.matters.news/graphql' from origin 'https://server.matters.news' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

**2** 
According to my test,
For `server-test` API, we didn't need `x-access-token` in header.
For `server` API, we need `x-access-token`  in header, but I can only get my likerid only.
You can test the two query yourself (with `x-access-token` in header) .
```
query {
  user(input: {userName:"guo"}) {
    liker {
      likerId
    }
  }
}
```
```
query {
  user(input: {userName:"askender"}) {
    liker {
      likerId
    }
  }
}
```

**Comment by guoliu** [comment]
> For server-test API, we didn't need x-access-token in header.
> For server API, we need x-access-token in header, but I can only get my likerid only.

Ah my bad. Yes, test environment is one step ahead of production environment. We have updated test environment, but it has not been deployed yet. You should be able to use production environment the same way next week after deployment.

**Comment by askender** [comment]
Hope there will be a deployment of this feature.
